app/controllers/application_controller.rb

```diff
@@ -6,4 +6,21 @@ class ApplicationController < ActionController::Base
   def favorite
     @favorite ||= Favorite.new(session[:favorite])
   end
+
+  def toggle_approval_status(pet_application)
+    if pet_application[:approval_status] == "unapproved"
+      "approved"
+    else
+      "unapproved"
+    end
+  end
+
+  def toggle_adoption_status(pet)
+    if pet.adoption_status == "adoptable"
+      "adoption pending"
+    else
+      "adoptable"
+    end
+  end
```

Comment:
I moved these here since they are being used in multiple controllers.

Reply:
Seems like a good move to make.